transhumandesign/kag-base
# Crate will change team to unpacker, Crate unpacking cannot be done from afar

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

When a crate is unpacked, it will change its team to the player who started unpacking.

When a player tries to click on "unpack" or "stop unpacking" buttons, he will have to be inside 20.0 radius. You cannot click those buttons from a screen length away anymore.

## Steps to Test or Reproduce

1. Go to Sandbox.
2. Switch to Blue Team.
3. !crate catapult
4. Pick up the crate and place it down. It should change to Blue Team.
5. Switch to Red Team.
6. Show buttons and click on "Unpack".
7. The crate will unpack a Blue Catapult. (After this change, it would instead be a Red Catapult).

---

1. Go to Sandbox.
2. !crate catapult
3. Keep showing "unpack" button and try to walk away.
4. Notice you can still click on the button even when you are a half screen length away. (After this change, you can only click on it if you are next to the crate.)
